apps/ledger-live-desktop/src/renderer/components/DeviceAction/rendering.tsx

```diff
@@ -1170,9 +1171,19 @@ const SwapDeviceConfirmation: React.FC<SwapConfirmationProps> = ({
   stateSettings,
   walletState,
 }) => {
-  const isDetailedViewEnabled = useFeature("ptxSwapDetailedView")?.enabled;
+  const ptxSwapDetailedView = useFeature("ptxSwapDetailedView");
+  const isDetailedViewEnabled = !!ptxSwapDetailedView?.enabled;
+  const variant = ptxSwapDetailedView?.params?.variant ?? ABTestingVariants.variantA;
   const sourceAccountCurrency = exchange.fromCurrency;
   const targetAccountCurrency = exchange.toCurrency;
+
+  useEffect(() => {
```

Comment:
before we add logic here to make sure we fire this event only once, I think it might make sense to just track the variant on the Page Tracking event, no?! Maybe check with Product people, but imho this would be the cleanest solution